tpm2/src/util/tpm2_kmyth_seal.c

```diff
@@ -665,6 +665,14 @@ int kmyth_wrap_input(char *inPath,
     return 1;
   }
   kmyth_log(LOG_DEBUG, "read in %d bytes of data to be wrapped", data_length);
+  
+  // validate non-empty plaintext buffer specified
+  if (data_length == 0)
```

Comment:
For completeness we should also check that `data` is not `NULL`.

Reply:
I apologize, my earlier comment wasn't clear. I meant that we should check that `data` is not `NULL` in addition to checking that `data_length` is not zero before using it, I wasn't suggesting to check that `data `is not `NULL` before `free`ing it.